src/node/internal/internal_http_server.ts

```diff
@@ -0,0 +1,708 @@
+  304: 'Not Modified', // RFC 7232 4.1
+  305: 'Use Proxy', // RFC 7231 6.4.5
+  307: 'Temporary Redirect', // RFC 7231 6.4.7
+  308: 'Permanent Redirect', // RFC 7238 3
+  400: 'Bad Request', // RFC 7231 6.5.1
+  401: 'Unauthorized', // RFC 7235 3.1
+  402: 'Payment Required', // RFC 7231 6.5.2
+  403: 'Forbidden', // RFC 7231 6.5.3
+  404: 'Not Found', // RFC 7231 6.5.4
+  405: 'Method Not Allowed', // RFC 7231 6.5.5
+  406: 'Not Acceptable', // RFC 7231 6.5.6
+  407: 'Proxy Authentication Required', // RFC 7235 3.2
+  408: 'Request Timeout', // RFC 7231 6.5.7
+  409: 'Conflict', // RFC 7231 6.5.8
+  410: 'Gone', // RFC 7231 6.5.9
+  411: 'Length Required', // RFC 7231 6.5.10
+  412: 'Precondition Failed', // RFC 7232 4.2
+  413: 'Payload Too Large', // RFC 7231 6.5.11
+  414: 'URI Too Long', // RFC 7231 6.5.12
+  415: 'Unsupported Media Type', // RFC 7231 6.5.13
+  416: 'Range Not Satisfiable', // RFC 7233 4.4
+  417: 'Expectation Failed', // RFC 7231 6.5.14
+  418: "I'm a Teapot", // RFC 7168 2.3.3
+  421: 'Misdirected Request', // RFC 7540 9.1.2
+  422: 'Unprocessable Entity', // RFC 4918 11.2
+  423: 'Locked', // RFC 4918 11.3
+  424: 'Failed Dependency', // RFC 4918 11.4
+  425: 'Too Early', // RFC 8470 5.2
+  426: 'Upgrade Required', // RFC 2817 and RFC 7231 6.5.15
+  428: 'Precondition Required', // RFC 6585 3
+  429: 'Too Many Requests', // RFC 6585 4
+  431: 'Request Header Fields Too Large', // RFC 6585 5
+  451: 'Unavailable For Legal Reasons', // RFC 7725 3
+  500: 'Internal Server Error', // RFC 7231 6.6.1
+  501: 'Not Implemented', // RFC 7231 6.6.2
+  502: 'Bad Gateway', // RFC 7231 6.6.3
+  503: 'Service Unavailable', // RFC 7231 6.6.4
+  504: 'Gateway Timeout', // RFC 7231 6.6.5
+  505: 'HTTP Version Not Supported', // RFC 7231 6.6.6
+  506: 'Variant Also Negotiates', // RFC 2295 8.1
+  507: 'Insufficient Storage', // RFC 4918 11.5
+  508: 'Loop Detected', // RFC 5842 7.2
+  509: 'Bandwidth Limit Exceeded',
+  510: 'Not Extended', // RFC 2774 7
+  511: 'Network Authentication Required', // RFC 6585 6
+};
+
+export type DataWrittenEvent = {
+  index: number;
+  entry: WrittenDataBufferEntry;
+};
+
+// @ts-expect-error TS2720 net.Server inconsistencies.
+export class Server
+  extends EventEmitter
+  implements _Server, BaseWithHttpOptions
+{
+  [kConnectionsCheckingInterval]?: number;
+  [kUniqueHeaders]: Set<string> | null = null;
+
+  // Similar option to this. Too lazy to write my own docs.
+  // http://www.squid-cache.org/Doc/config/half_closed_clients/
+  // https://wiki.squid-cache.org/SquidFaq/InnerWorkings#What_is_a_half-closed_filedescriptor.3F
+  httpAllowHalfOpen = false;
+  timeout = 0;
+  maxHeadersCount: number | null = null;
+  maxRequestsPerSocket = 0;
+
+  requestTimeout: number = 0;
+  headersTimeout: number = 0;
+  requireHostHeader: boolean = false;
+  joinDuplicateHeaders: boolean = false;
+  rejectNonStandardBodyWrites: boolean = false;
+  keepAliveTimeout: number = 5_000;
+  port?: number;
```

Comment:
I may have missed it but it appears this is not implementing the `server.address()` function? It is quite common for code, particularly logging code, to call `server.address().port`, etc, from within the `'listening'` event handler. Even if it doesn't quite work in our environment we should probably provide it. https://nodejs.org/docs/latest/api/net.html#serveraddress